code/src/App.js

```diff
@@ -1,29 +1,51 @@
-import React from "react";
-import { BarcodeScanner } from "components/BarcodeScanner";
+import React from 'react'
+import { Provider } from 'react-redux'
+import { configureStore, combineReducers } from '@reduxjs/toolkit'
+import { Product } from 'components/Product'
+import { LoadingIndicator } from 'components/LoadingIndicator'
+import { products } from './reducers/products'
+import { loading } from './reducers/loading'
+import styled from 'styled-components/macro'
 
-const onDetected = (code) => {
-  console.log(`Code: ${code}`);
-  fetch(`https://world.openfoodfacts.org/api/v0/product/${code}.json`)
-    .then((data) => data.json())
-    .then((json) => {
-      console.log(json);
-    });
-};
+const reducer = combineReducers({
+  loading: loading.reducer,
+  products: products.reducer
+})
+
+export const store = configureStore({ reducer })
 
 export const App = () => {
   return (
-    <div>
-      <label>
-        {" "}
-        Test codes here:{" "}
-        <input type="text" onChange={(e) => onDetected(e.target.value)}></input>
-      </label>
-      <p>
-        {" "}
-        Use the field above to test barcodes manually and keep an eye on your
-        console in the browser. i.e. Type 7311070347272 - Pågen Gifflar. Yum
-      </p>
-      <BarcodeScanner onDetected={onDetected}></BarcodeScanner>
-    </div>
-  );
-};
+    <Provider store={store}>
+      <MainContainer>
+        <LoadingIndicator />
+        <h1>Is this Cheese?</h1>
+        <ProductContainer />
+      </MainContainer>
+    </Provider>
+  )
+}
+
+const MainContainer = styled.section`
```

Comment:
Your styled components should be defined above the actual react component. It works this way, but it's best practice to define things before they're used (reading from top to bottom)